sv_mini_atlas/tests/test_importer.py

```diff
@@ -243,169 +233,150 @@ def test_importer(mock_node, mock_open):
     new_callable=mock.mock_open,
     read_data=constants.PASSAGE,
 )
+@mock.patch("sv_mini_atlas.library.importers.CTSImporter.generate_node")
 @mock.patch("sv_mini_atlas.library.importers.Node")
-def test_importer_exemplar(mock_node, mock_open):
+def test_importer_with_exemplar(mock_node, mock_generate, mock_open):
     version_urn = "urn:cts:greekLit:tlg0012.tlg001.perseus-grc2:"
     exemplar_urn = "urn:cts:greekLit:tlg0012.tlg001.perseus-grc2.card:"
-
     library_ = copy.deepcopy(library)
     exemplar_data = library_.versions.pop(version_urn)
     exemplar_data.update({"urn": exemplar_urn})
     library_.versions[exemplar_urn] = exemplar_data
+
     CTSImporter(library_, exemplar_data, {}).apply()
 
-    assert mock_node.mock_calls == [
-        mock.call.add_root(kind="nid", urn="urn:cts:", idx=0),
-        mock.call.add_root().add_child(
-            kind="namespace", urn="urn:cts:greekLit:", idx=0
+    assert mock_generate.mock_calls == [
+        mock.call(0, {"kind": "nid", "urn": "urn:cts:", "idx": 0}, None),
+        mock.call(
+            1, {"kind": "namespace", "urn": "urn:cts:greekLit:", "idx": 0}, "urn:cts:"
+        ),
+        mock.call(
+            2,
+            {"kind": "textgroup", "urn": "urn:cts:greekLit:tlg0012:", "idx": 0},
+            "urn:cts:greekLit:",
+        ),
+        mock.call(
+            3,
+            {"kind": "work", "urn": "urn:cts:greekLit:tlg0012.tlg001:", "idx": 0},
+            "urn:cts:greekLit:tlg0012:",
         ),
```

Comment:
nice; this is a lot easier to reason about!

Reply:
Yes, I was pleased about that too.